### PR TITLE
SAT implementation 

### PR DESCRIPTION
**Pull request by jmgomez**
Disabled tests will be added in a future PR

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -32,6 +32,7 @@ jobs:
         run: nimble install -y
       - name: Run nim c -r tester
         run: |
+          nimble install
```
Comment (ringabout) [inline .github/workflows/test.yml:35]:
```suggestion
```

File: src/nimble.nim [hunk 2]
```diff
@@ -8,17 +8,17 @@ import std/options as std_opt
 
 import strutils except toLower
 from unicode import toLower
-
+import sat/sat
```
Comment (ringabout) [inline src/nimble.nim:11]: Now or later, you need to ensure `koch` bundles `nimble` correctly, which needs to clone `sat` instead.

File: .github/workflows/test.yml (hunk 1 above)
Comment (jmgomez) [reply .github/workflows/test.yml:35]: It needs `nimble install` for running the tests, no?

Comment (ringabout) [reply .github/workflows/test.yml:35]: > run: nimble install -y

My previous PR had added it already above

**Pull request merged**